platform/server/elasticsearch/ElasticsearchService.ts

```diff
@@ -93,8 +94,9 @@ export class ElasticsearchService implements CoreService {
   // This method is really transparent. It does the two things
   // we currently do. I think we could work Observables
   // into this so we get updated stuff.
-  getScopedCluster(type: ElasticsearchClusterType, headers: string): Cluster {
-    const { callWithRequest } = this.getCluster(type);
-    return (...args) => this.callWithRequest(headers, ...args);
+  async getScopedCluster(type: ElasticsearchClusterType, headers: Headers): Promise<Cluster> {
```

Comment:
You don't need to specify `: Promise<Cluster>` here, as it's already inferred from `getClusterOfType`

Reply:
Oh, I thought I removed these comments related to TS. Don't worry about them for now (it's better to focus on the stuff I mention in the huge comment below)